R/dummy_functions.R

```diff
@@ -36,30 +45,46 @@ example_module <- function(label = "example teal module", datanames = "all", tra
           )
         })
 
+        table_data <- reactive({
+          within(data(),
+            {
+              object <- dataname
+            },
+            dataname = as.name(input$dataname)
+          )
+        })
+
+        table_data_decorated <- srv_teal_transform_data("decorate", data = table_data, transformators = decorators)
```

Comment:
I think the decorator's expression is being triggered multiple times here and we should limit it to only when data changes. Even when the module is not being active.

[Screencast from 2024-11-18 12-40-17.webm](https://github.com/user-attachments/assets/7c7946a6-324e-48fe-9f61-87b0a18ae621)


Try with:

```r
pkgload::load_all("teal")

options(
  teal.log_level = "TRACE",
  teal.show_js_log = TRUE,
  # teal.bs_theme = bslib::bs_theme(version = 5),
  shiny.bookmarkStore = "server"
)

output_decorator <- teal_transform_module(
  label = "a label",
  server = make_teal_transform_server(
    expression({
      print("Expression being called!")
      object <- rev(object)
    })
  )
)

app <- init(
  data = teal.data::teal_data(data1 = iris, data2 = mtcars),
  modules = modules(
    example_module("module 1", decorators = output_decorator),
    example_module("module (no decorator)")
  )
)
if (interactive()) {
  shinyApp(app$ui, app$server)
}
```

Reply:
@averissimo I think this is the same as we see here in the vignette https://github.com/insightsengineering/teal/pull/1357#discussion_r1842490188

Reply:
It is. I'm resolving this one and discuss it there.

thanks @m7pr :heart:

Reply:
So `srv_teal_transform_data` should be called only if `table_data` is changed?